Realm/Realm.Sync/SyncConfiguration.cs

```diff
@@ -78,15 +79,24 @@ internal override Realm CreateRealm(RealmSchema schema)
                 schema_version = SchemaVersion
             };
 
-            var syncConfiguration = new Native.SyncConfiguration
+            var srHandle = SharedRealmHandleExtensions.OpenWithSync(configuration, ToNative(), schema, EncryptionKey);
+            return new Realm(srHandle, this, schema);
+        }
+
+        internal override async Task<Realm> CreateRealmAsync(RealmSchema schema)
+        {
+            var session = new Session(SharedRealmHandleExtensions.GetSession(DatabasePath, ToNative(), EncryptionKey));
+            await session.WaitForDownloadAsync();
+            return CreateRealm(schema);
```

Comment:
I honestly have no idea, but I imagine that's something that will be solved at the Object Store level.